exec/vector/src/main/java/org/apache/drill/exec/vector/accessor/reader/AbstractTupleReader.java

```diff
@@ -87,6 +87,13 @@ public void bindNullState(NullStateReader nullStateReader) {
     this.nullStateReader = nullStateReader;
   }
 
+  @Override
+  public void rebind() {
+    for (int i = 0; i < readers.length; i++) {
+      readers[i].events().rebind();
+    }
```

Comment:
It can be rewritten with a for-each loop:
```
    for (AbstractObjectReader reader : readers) {
      reader.events().rebind();
    }
```
There is no difference from the performance perspective, but it would be easier to support this code in the future, for example when replacing array with a list, etc.

Reply:
Generally I do use the iterator form. However, the readers and writers are inner-loop code: old-fashioned index looping is faster and avoids creating heap objects. The code here was specifically designed for index looping.

Reply:
Actually, at least Java 8 rewrites for-each loop for array into the for-i loop, here is an example of the source code and decompiled code:
```
    String[] arr = {"a", "b", "c"};
    for (String x : arr) {
      System.out.println(x);
    }
```
code after decompilation:
```
    String[] var1 = new String[]{"a", "b", "c"};
    String[] var2 = var1;
    int var3 = var1.length;

    for(int var4 = 0; var4 < var3; ++var4) {
      String var5 = var2[var4];
      System.out.println(var5);
    }
```
But anyway, this is a minor comment, so the code may be left as it is.